core/store/orm/orm.go

```diff
@@ -326,6 +339,44 @@ func (orm *ORM) LinkEarnedFor(spec *models.JobSpec) (*assets.Link, error) {
 	return earned, nil
 }
 
+// UpsertErrorFor upserts a JobSpecError record, incrementing the occurences counter by 1
+// if the record is found
+func (orm *ORM) UpsertErrorFor(jobID *models.ID, description string) {
+	orm.MustEnsureAdvisoryLock()
+	err := orm.db.Exec(`
+		INSERT INTO job_spec_errors (
+			job_spec_id, description, created_at, updated_at
+		) VALUES (
+				?, ?, ?, ?
+		) ON CONFLICT (job_spec_id, description)
+		DO UPDATE
+		SET occurences = job_spec_errors.occurences + 1, updated_at = ?
+`, jobID, description, time.Now(), time.Now(), time.Now()).Error
```

Comment:
This will give you three different times, probably not what you want. You may like to use the built-in SQL function `NOW()` instead (this gives a consistent time inside a transaction).

Reply:
ohhhh good call - that's so much better

Reply:
going with above recommendation, but still good to know